backend/parsers/audio.py

```diff
@@ -43,33 +44,40 @@ async def process_audio(commons: CommonsDep, upload_file: UploadFile, enable_sum
     if user_openai_api_key:
         openai_api_key = user_openai_api_key
 
-    # Here, we're writing the uploaded file to a temporary file, so we can use it with your existing code.
-    with tempfile.NamedTemporaryFile(delete=False, suffix=upload_file.filename) as tmp_file:
-        await upload_file.seek(0)
-        content = await upload_file.read()
-        tmp_file.write(content)
-        tmp_file.flush()
-        tmp_file.close()
+    try:
+        # Here, we're writing the uploaded file to a temporary file, so we can use it with your existing code.
+        with tempfile.NamedTemporaryFile(delete=False, suffix=upload_file.filename) as tmp_file:
+            await upload_file.seek(0)
+            content = await upload_file.read()
+            tmp_file.write(content)
+            tmp_file.flush()
+            tmp_file.close()
 
-        with open(tmp_file.name, "rb") as audio_file:
-            transcript = openai.Audio.transcribe("whisper-1", audio_file)
+            temp_filename = tmp_file.name
 
-    file_sha = compute_sha1_from_content(transcript.text.encode("utf-8"))
-    file_size = len(transcript.text.encode("utf-8"))
+            with open(tmp_file.name, "rb") as audio_file:
+                transcript = openai.Audio.transcribe("whisper-1", audio_file)
 
-    # Load chunk size and overlap from sidebar
-    chunk_size = 500
-    chunk_overlap = 0
+        file_sha = compute_sha1_from_content(transcript.text.encode("utf-8"))
+        file_size = len(transcript.text.encode("utf-8"))
 
-    text_splitter = RecursiveCharacterTextSplitter.from_tiktoken_encoder(
-        chunk_size=chunk_size, chunk_overlap=chunk_overlap)
-    texts = text_splitter.split_text(transcript)
+        # Load chunk size and overlap from sidebar
+        chunk_size = 500
+        chunk_overlap = 0
 
-    docs_with_metadata = [Document(page_content=text, metadata={"file_sha1": file_sha, "file_size": file_size, "file_name": file_meta_name,
-                                   "chunk_size": chunk_size, "chunk_overlap": chunk_overlap, "date": dateshort}) for text in texts]
+        text_splitter = RecursiveCharacterTextSplitter.from_tiktoken_encoder(
+            chunk_size=chunk_size, chunk_overlap=chunk_overlap)
+        texts = text_splitter.split_text(transcript)
```

Comment:
When attempting to debug locally, it appears that the correct statement should be `texts = text_splitter.split_text(transcript.text)` instead of `texts = text_splitter.split_text(transcript)`.

This could potentially be the fix we're looking for

Reply:
That is true. The default response format is JSON for the transcription API, unless you explicitly specify text in the request parameter. 

Added this fix too.

Reply:
That said, while this fixes the bug, it is not a robust feature until we have a solution around the [25MB file size limit](https://help.openai.com/en/articles/7031512-whisper-api-faq#:~:text=size%20is%20supported%3F-,Up%20to%2025MB,-.).

Reply:
Agreed. We should consider implementing a chunk system.

Reply:
And utilise PyDub to split on silence so it doesn't mess up the transcription if we chunk it in the middle of a word or something.